File: jsParser/parser/While_statementVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const While_statement = require('../domain/While_statement')

const StatementVisitor = require('./StatementVisitor')
const ExpressionVisitor = require('./ExpressionVisitor')

class While_statementVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitWhile_statement (ctx) {
    let expression = null
    const loop_statements = []

    if (ctx.expression()) {
      const expressionVisitor = new ExpressionVisitor.ExpressionVisitor()
      expression = expressionVisitor.visitExpression(ctx.expression())
    }

    if (ctx.statement()) {
      const statementVisitor = new StatementVisitor.StatementVisitor()
      ctx.statement().forEach(stmt => {
        loop_statements.push(statementVisitor.visitStatement(stmt))
      })
    }

    return new While_statement.While_statement(expression, loop_statements)
  }
};

While_statementVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitWhile_statement = this.visitWhile_statement
exports.While_statementVisitor = While_statementVisitor

File: java/parser/src/main/java/gov/lbl/parser/parser/While_statementVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Expression;
import gov.lbl.parser.domain.Statement;
import gov.lbl.parser.domain.While_statement;

import static java.util.stream.Collectors.toList;

public class While_statementVisitor  extends modelicaBaseVisitor<While_statement> {
    @Override
    public While_statement visitWhile_statement(modelicaParser.While_statementContext ctx) {
        ExpressionVisitor expressionVisitor = new ExpressionVisitor();
        Expression expression = ctx.expression() == null ? null : ctx.expression().accept(expressionVisitor);

        StatementVisitor statementVisitor = new StatementVisitor();
        List<Statement> loop_statements = ctx.statement() == null ? null : ctx.statement()
                                                                                .stream()
                                                                                .map(stmt -> stmt.accept(statementVisitor))
                                                                                .collect(toList());
        return new While_statement(expression, loop_statements);
    }
}